webportal/src/app/job/job-view/job-view.component.js

```diff
@@ -147,20 +147,45 @@ const loadJobs = () => {
       if (data.error) {
         alert(data.message);
       } else {
-        $('#view-table').html(jobTableComponent({
-          jobs: data.slice(0, 300),
-          getDurationInSeconds,
-          convertTime,
-          convertState,
-        }));
+        let displayDataSet = [];
+        let rowCount = limit ? limit : 1000;
+        for (let i = 0; i < data.slice(0, rowCount).length; i++) {
```

Comment:
What if rowCount < 0?

Reply:
Can we use this logic:
- If `limit` is not specified, we display all the jobs.
- If `limit` is specified as a non-negative number, we display at most `limit` number of jobs.
- If `limit` is specified as a negative number, we ignore it (as if `limit` is not specified).

With this change, the default link of the 'Jobs' menu item on the left pane of the web portal should be `/view.html?limit=1000`.

Reply:
if limit is not specified, shouldn't we display 1000 jobs by default?

Reply:
No. Because it's not intuitive.